src/main/scala/io/iohk/ethereum/vm/PrecompiledContracts.scala

```diff
@@ -65,15 +65,23 @@ object PrecompiledContracts {
       val r = data.slice(64, 96)
       val s = data.slice(96, 128)
 
-      val recovered = Try(ECDSASignature(r,s,v).publicKey(h)).getOrElse(None)
-      recovered.map { bytes =>
-        val hash = kec256(bytes).slice(12, 32)
-        ByteUtils.padLeft(hash, 32)
-      }.getOrElse(ByteString.empty)
+      if (hasOnlyLastByteSet(v)) {
+        val recovered = Try(ECDSASignature(r, s, v.last).publicKey(h)).getOrElse(None)
+        recovered.map { bytes =>
+          val hash = kec256(bytes).slice(12, 32)
+          ByteUtils.padLeft(hash, 32)
+        }.getOrElse(ByteString.empty)
+      } else
+        ByteString.empty
+
     }
 
     def gas(inputDataSize: UInt256): BigInt =
       3000
+
+    private def hasOnlyLastByteSet(v: ByteString): Boolean = {
+      v.dropWhile(_ == 0).size == 1
+    }
```

Comment:
Minor, I suggest we skip the braces: https://docs.scala-lang.org/style/declarations.html#body